raiden/transfer/mediated_transfer/target.py

```diff
@@ -232,30 +233,33 @@ def handle_block(
     """
     transfer = target_state.transfer
     events = list()
+    lock = transfer.lock
 
     secret_known = channel.is_secret_known(
         channel_state.partner_state,
-        transfer.lock.secrethash,
+        lock.secrethash,
     )
-    is_lock_expired = (
-        not secret_known and
-        block_number > transfer.lock.expiration
+    lock_has_expired, lock_expired_msg = channel.is_lock_expired(
+        end_state=channel_state.our_state,
+        lock=lock,
+        block_number=block_number,
+        lock_expiration_threshold=lock.expiration + DEFAULT_NUMBER_OF_CONFIRMATIONS_BLOCK,
```

Comment:
Is it intentional that we also don't wait for 2*block_confirmations?

Reply:
this is intentional, the receiver waits for the confirmation blocks to guarantee that if the secret was registered onchain then it will know, the sender waits for these confirmation blocks because the receiver will reject the message, but the sender also waits for an some additional blocks to account for the receiver being behind (it could be a value different then confirmation blocks)